File: haskell/src/AtCoder/Beginner/Q110/A.hs
module AtCoder.Beginner.Q110.A where
import Control.Monad
import Control.Applicative
import Data.List
import qualified Data.Map.Strict as M
import Control.Monad (mfilter)

toInt :: (String -> Int)
toInt x = read x ::Int

getInts = do
  x <- getLine
  let y = (map$toInt)$words x
  return y

getStr = do
  x <- getLine
  let y = words x
  return y

arrayToInt :: [Int] -> Int
arrayToInt a = foldl (+) 0 $ map (\x -> snd x * 10^(fst x)) $ zip [0..((length a)-1)] a

q1 :: ([Int], [Int]) -> Int
q1 x = (arrayToInt . snd) x + (head . fst) x

answer :: IO ()
answer = do
  x <- getInts
  print $ q1 $ splitAt 1 $ sort x

File: scala/src/main/scala/atcoder/beginner/q110/A.scala
package atcoder.beginner.q110

import scala.io.StdIn.readLine

object A {
  def input[A](change: String => A) = readLine.split(" ").map(change(_))
  def main(args: Array[String]): Unit ={
    val N = input((a=>a.toInt))
    val M = N.sortBy(a=>a)
    println(M.tail.foldLeft(M.head, 1)((acc, c) => (acc._1+c*acc._2, acc._2*10))._1)
  }
}
